MolkPackageQueue/PriorityQueue.cs

```diff
@@ -8,17 +8,82 @@ namespace MolkPackageQueue
 {
     public class PriorityQueue
     {
+
         Queue<Package> queueHigh = new Queue<Package>();
         Queue<Package> queueMedium = new Queue<Package>();
         Queue<Package> queueLow = new Queue<Package>();
+        List<Package> outgoingPackages = new List<Package>();
 
+        // Enqueue packages based on its priority
         public void Enqueue(Package package)
         {
-            // do stuff
+            if(package.Priority == Priority.High)
+                queueHigh.Enqueue(package);
+            if(package.Priority == Priority.Low)
+                queueLow.Enqueue(package);
+            if(package.Priority == Priority.Medium)
+                queueMedium.Enqueue(package);
+        }
+
+        // Dequeue random number of packages every time its called, removes a system check varibles value by number of dequeu packages
+        public void Dequeue()
+        {
+            Random random = new Random();
+            int numberOfDequeu = 0;
+
+            while (numberOfDequeu < random.Next(1, 6))
+            {
+                if (queueHigh.Count > 0)
+                {
+                    Package packages = queueHigh.Dequeue();
+                    AddProccesPackage(packages);
+                    numberOfDequeu++;
+                    Program.numberOfOrderProcessed--;
+                }
+                else if (queueMedium.Count > 0)
+                {
+                    Package packages = queueMedium.Dequeue();
+                    AddProccesPackage(packages);
+                    numberOfDequeu++;
+                    Program.numberOfOrderProcessed--;
+                }
+                else if (queueLow.Count > 0)
+                {
+                    Package packages = queueLow.Dequeue();
+                    AddProccesPackage(packages);
+                    numberOfDequeu++;
+                    Program.numberOfOrderProcessed--;
+                }
```

Comment:
if random.Next gives a high number in the start of this and there is less packages in all queues this will loop alot of loops since numberOfDequeue never increases.  So you are two randoms from a eternal loop.